lib/web/router/impl.ex

```diff
@@ -106,7 +106,7 @@ defmodule Antikythera.Router.Impl do
            action :: v[atom],
            opts :: Keyword.t(any)
          ) :: Macro.t() do
-    websocket? = !!opts[:websocket?]
+    websocket? = opts[:websocket?] || false
```

Comment:
IIUC, credo says "`!!value` is difficult to understand whether `nil` for `false` is bug".
https://hexdocs.pm/credo/Credo.Check.Refactor.DoubleBooleanNegation.html

I think `not !value` is difficult to understand for the same reason.

As the above page says, I think we should create a small function for this.

```elixir
defp is_route_for_websocket?(nil), do: false
defp is_route_for_websocket?(false), do: false
defp is_route_for_websocket?(_), do: true
```

It's ok to create a function which gets `opts` instead of `opts[:websocket?]`

If you think it's too long for this simple code, we can simply disable this warning for this line and add a comment like the following.
> The caller passes `true` or `nil` for `opts[:websocket?]`

Reply:
[API document of Antikythera.Router](https://hexdocs.pm/antikythera/Antikythera.Router.html) says that only `:from` and `:as` options are allowed as the 4th argument of routing macros. `:websocket?` option is basically for internal use. We don't have to worry about getting non-boolean value from `opts[:websocket?]`.

BTW, this code might be better for readability:
```elixir
# The value of option `:websocket?` must be boolean or nothing.
websocket? = Keyword.get(opts, :websocket?, false)
```

It is also helpful to validate the value of `websocket?` afterward:
```elixir
unless is_boolean(websocket?) do
  raise "option `:websocket?` must be boolean but given: #{websocket?}"
end
```

Reply:
Fixed at 0b8527b and 6a07b82.